src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -204,33 +202,48 @@ private SingleTypeDeclaration FirstDeclarationWithExplicitBases()
             return null;
         }
 
-        internal Tuple<NamedTypeSymbol, ImmutableArray<NamedTypeSymbol>> GetDeclaredBases(ConsList<Symbol> basesBeingResolved)
+        internal Tuple<NamedTypeSymbol, ImmutableArray<NamedTypeSymbol>> GetDeclaredBases(ConsList<Symbol> basesBeingResolved, bool ignoreNonNullTypesAttribute)
         {
             if (ReferenceEquals(_lazyDeclaredBases, null))
             {
-                DiagnosticBag diagnostics = DiagnosticBag.GetInstance();
-                if (Interlocked.CompareExchange(ref _lazyDeclaredBases, MakeDeclaredBases(basesBeingResolved, diagnostics), null) == null)
+                if (ignoreNonNullTypesAttribute)
                 {
-                    AddDeclarationDiagnostics(diagnostics);
+                    if (ReferenceEquals(_lazyDeclaredBasesIgnoringNonNullTypesAttribute, null))
+                    {
+                        initDeclaredBases(ref _lazyDeclaredBasesIgnoringNonNullTypesAttribute);
+                    }
+
+                    return _lazyDeclaredBasesIgnoringNonNullTypesAttribute;
                 }
 
-                diagnostics.Free();
+                initDeclaredBases(ref _lazyDeclaredBases);
             }
 
             return _lazyDeclaredBases;
+
+            void initDeclaredBases(ref Tuple<NamedTypeSymbol, ImmutableArray<NamedTypeSymbol>> bag)
+            {
+                var diagnostics = DiagnosticBag.GetInstance();
+                if (Interlocked.CompareExchange(ref bag, MakeDeclaredBases(basesBeingResolved, ignoreNonNullTypesAttribute, diagnostics), null) == null)
+                {
+                    AddDeclarationDiagnostics(diagnostics);
```

Comment:
>AddDeclarationDiagnostics(diagnostics) [](start = 20, length = 38)

Are we reporting the same diagnostics twice?

Reply:
There was no problem of duplicate diagnostics because the caller that passes `ignoreNonNullTypesAttributes` is `GetNextBaseTypeNoUseSiteDiagnostics` which ignores the diagnostics. 
However I think you're observation is right, so I removed that logic. Thanks

---
In reply to: [201404641](https://github.com/dotnet/roslyn/pull/28320#discussion_r201404641) [](ancestors = 201404641)